llvm_util/utils.cpp

```diff
@@ -41,10 +41,34 @@ ostream *out;
 
 const llvm::DataLayout *DL;
 
+static void _getUsedGlobalVariables(llvm::User *U,
+    llvm::SmallPtrSet<llvm::Value *, 16> &Visited,
+    llvm::SmallPtrSet<llvm::GlobalVariable *, 16> &GlobalVars) {
+  Visited.insert(U);
+  for (auto I = U->op_begin(), E = U->op_end(); I != E; ++I) {
+    llvm::Value *V = I->get();
+    if (Visited.count(V)) continue;
+    else if (llvm::GlobalVariable *GV = dyn_cast<llvm::GlobalVariable>(V)) {
+      GlobalVars.insert(GV);
+    }
+  }
+}
+
 }
 
 namespace llvm_util {
 
+vector<llvm::GlobalVariable *> getUsedGlobalVariables(llvm::Function &F) {
```

Comment:
uhm, why is this analysis implicit when generating alive code? We already traverse the whole code once, why this additional traversal?

Reply:
The reason was that making get_operand do these thing seemed to make get_operand too complex.
When constexprs are visited at initializers, corresponding instructions are generated at the separate basic block unlike constexprs in a function, and creation of the store to the global variable is needed.
Instead of making get_operand more complex (which was already carrying constexpr_conv around), I put the initializer generation separately before beginning of function translation, which required this analysis.

Reply:
Or alternatively, we can move get_operand to llvm2alive.cpp.
This way, get_operand wouldn't need to carry auxiliaries needed for constexprs.